test/ArrayList.test.js

```diff
@@ -0,0 +1,445 @@
+        const expectedString = "[1]";
+        aList.init(arr);
+
+        const actual = aList.toString();
+
+        assert.deepEqual(actual, expectedString);
+    });
+
+    it('should return "[1, 2]" [1, 2]', () => {
+        const aList = new ArrayList();
+        const arr = [1, 2];
+        const expectedString = "[1, 2]";
+        aList.init(arr);
+
+        const actual = aList.toString();
+
+        assert.deepEqual(actual, expectedString);
+    });
+
+    it('should return "[1, 2, 4, -1, 3]" [1, 2, 4, -1, 3]', () => {
+        const aList = new ArrayList();
+        const arr = [1, 2, 4, -1, 3];
+        const expectedString = "[1, 2, 4, -1, 3]";
+        aList.init(arr);
+
+        const actual = aList.toString();
+
+        assert.deepEqual(actual, expectedString);
+    });
+});
+
+describe('ArrayList.push', () => {
+    it('should return 0 (undefined, undefined)', () => {
+        const aList = new ArrayList();
+        const arr = undefined;
+        const value = undefined;
+        const expectedString = "[]";
+        const expected = 0;
+        aList.init(arr);
+
+        const actual = aList.push(value);
+
+        assert.deepEqual(aList.toString(), expectedString);
+        assert.deepEqual(actual, expected);
+    });
+
+    it('should return 1 (undefined, 19)', () => {
+        const aList = new ArrayList();
+        const arr = [];
+        const value = 19;
+        const expectedString = "[19]" ;
+        const expected = 1;
+        aList.init(arr);
+
+        const actual = aList.push(value);
+
+        assert.deepEqual(aList.toString(), expectedString);
+        assert.deepEqual(actual, expected);
+    });
+
+    it('should return 2 ([1], 19)', () => {
+        const aList = new ArrayList();
+        const arr = [1];
+        const value = 19;
+        const expected = 2;
+        const expectedString = "[1, 19]";
+        aList.init(arr);
+
+        const actual = aList.push(value);
+
+        assert.deepEqual(aList.toString(), expectedString);
+        assert.deepEqual(actual, expected);
+    });
+
+    it('should return 3 ([1, 2], 19)', () => {
+        const aList = new ArrayList();
+        const arr = [1, 2];
+        const value = 19;
+        const expected = 3;
+        const expectedString = "[1, 2, 19]";
+        aList.init(arr);
+
+        const actual = aList.push(value);
+
+        assert.deepEqual(aList.toString(), expectedString);
+        assert.deepEqual(actual, expected);
+    });
+
+    it('should return 6 ([1, 2, 4, -1, 3], 19)', () => {
+        const aList = new ArrayList();
+        const arr = [1, 2, 4, -1, 3];
+        const value = 19;
+        const expected = 6;
+        const expectedString = "[1, 2, 4, -1, 3, 19]";
+        aList.init(arr);
+
+        const actual = aList.push(value);
+
+        assert.deepEqual(aList.toString(), expectedString);
+        assert.deepEqual(actual, expected);
+    });
+});
+
+describe('ArrayList.pop', () => {
+    it('should return 0 ([19])', () => {
```

Comment:
no. If you have empty ArrayList, initialized with undefined or [], you still can call alist.pop(). It should simply return undefined in these cases. No such cases seen